extensions/amp-story-panning-media/0.1/test/test-amp-story-panning-media.js

```diff
@@ -61,31 +68,40 @@ describes.realWin(
 
     beforeEach(() => {
       win = env.win;
+
+      storeService = new AmpStoryStoreService(win);
+      registerServiceBuilder(win, 'story-store', function () {
+        return storeService;
+      });
     });
 
     it('should build', async () => {
       await createAmpStoryPanningMedia(
         '/examples/amp-story/img/conservatory-coords.jpg'
       );
-      return expect(() => panningMedia.layoutCallback()).to.not.throw();
+      expect(() => panningMedia.layoutCallback()).to.not.throw();
     });
 
     it('should throw if nested amp-img is missing', async () => {
       await createAmpStoryPanningMedia();
-      return expect(() => panningMedia.layoutCallback()).to.throw(
-        'Element expected: null'
-      );
+      expect(() =>
+        allowConsoleError(() => panningMedia.layoutCallback())
+      ).to.throw('Element expected: null');
     });
 
-    it('sets transform of image element from attributes', async () => {
+    it('sets transform of amp-img on page change', async () => {
       const positionValues = {x: '50%', y: '50%', zoom: '2'};
+
       await createAmpStoryPanningMedia(
         '/examples/amp-story/img/conservatory-coords.jpg',
         positionValues
       );
       await panningMedia.layoutCallback();
-      expect(panningMedia.image_.style.transform).to.equal(
-        `scale(${positionValues.zoom}) translate(${positionValues.x}, ${positionValues.y})`
+      await storeService.dispatch(Action.CHANGE_PAGE, {id: 'page1', index: 0});
+      requestAnimationFrame(() =>
```

Comment:
nit: extract the `rAF` to a promise that wraps it like in here:
https://github.com/ampproject/amphtml/blob/101659c52ef1c5c6f9e675c02a6423d4afeb0952/extensions/amp-base-carousel/0.1/test/test-child-layout-manager.js#L25

and then just:

```javascript
await afterRenderPromise()
expect()...
```

Reply:
Super clean. I like how it follows the visual pattern of our tests and is readable,